### ISSUE TITLE
Model Training Stalls: No Change in Accuracy or Evaluation Metrics on pFedMe

### ISSUE THREAD

**Issue by dDCTRr** [TsingZ0/PFL-Non-IID#118]
**ATTENTION : this problem only happen on the pFedMe method**

Abstract : There was an issue with the model training process, and the accuracy never improved.

Details:

steps to reproduce the problem:
```sh
git clone https://github.com/TsingZ0/PFL-Non-IID.git
cd PFL-Non-IID
cd dataset
python generate_cifar10.py noniid - dir
cd ../system
python main.py -did 0 -data Cifar10 -nb 10 -m cnn -lbs 16 -gr 50 -ls 10 -algo pFedMe -jr 1 -nc 10
```

and get the output as follows:
"

Algorithm: pFedMe
Local batch size: 16
Local steps: 10
Local learing rate: 0.005
Local learing rate decay: False
Total number of clients: 10
Clients join in each round: 1.0
Clients randomly join: False
Client drop rate: 0.0
Client select regarding time: False
Running times: 1
Dataset: Cifar10
Number of classes: 10
Backbone: cnn
Using device: cuda
Using DP: False
Auto break: False
Global rounds: 50
Cuda device id: 0
DLG attack: False
Total number of new clients: 0
Fine tuning epoches on new clients: 0

============= Running time: 0th =============
Creating server and clients ...
FedAvgCNN(
  (conv1): Sequential(
    (0): Conv2d(3, 32, kernel_size=(5, 5), stride=(1, 1))
    (1): ReLU(inplace=True)
    (2): MaxPool2d(kernel_size=(2, 2), stride=(2, 2), padding=0, dilation=1, ceil_mode=False)
  )
  (conv2): Sequential(
    (0): Conv2d(32, 64, kernel_size=(5, 5), stride=(1, 1))
    (1): ReLU(inplace=True)
    (2): MaxPool2d(kernel_size=(2, 2), stride=(2, 2), padding=0, dilation=1, ceil_mode=False)
  )
  (fc1): Sequential(
    (0): Linear(in_features=1600, out_features=512, bias=True)
    (1): ReLU(inplace=True)
  )
  (fc): Linear(in_features=512, out_features=10, bias=True)
)

Join ratio / total clients: 1.0 / 10
Finished creating server and clients.

-------------Round number: 0-------------

Evaluate personalized model
Average Test Accurancy: 0.1092
Average Test AUC: 0.1091
Average Train Loss: 2.3007

-------------Round number: 1-------------

Evaluate personalized model
Average Test Accurancy: 0.1092
Average Test AUC: 0.1091
Average Train Loss: 2.3007

-------------Round number: 2-------------

Evaluate personalized model
Average Test Accurancy: 0.1092
Average Test AUC: 0.1091
Average Train Loss: 2.3007

-------------Round number: 3-------------

Evaluate personalized model
Average Test Accurancy: 0.1092
Average Test AUC: 0.1091
Average Train Loss: 2.3007

-------------Round number: 4-------------

Evaluate personalized model
Average Test Accurancy: 0.1092
Average Test AUC: 0.1091
Average Train Loss: 2.3007

-------------Round number: 5-------------

Evaluate personalized model
Average Test Accurancy: 0.1092
Average Test AUC: 0.1091
Average Train Loss: 2.3007

-------------Round number: 6-------------

Evaluate personalized model
Average Test Accurancy: 0.1092
Average Test AUC: 0.1091
Average Train Loss: 2.3007

.......and so on
"

it seems that the accuracy as well as other evaluation index never rises or even changes

**Comment by dDCTRr** [comment]
sorry i found that's mostly perhaps caused by my awful hyperparameter selection.

**Comment by mahendrathapa** [comment]
@xiyuanyang45 Are you able to find the good hyperparameters?
